### PR TITLE
Add clippy checks for examples and tests.

### PR DESCRIPTION
**Pull request by xStrom**
The current CI configuration wasn't checking examples and tests for clippy mistakes.

### CODE REVIEW

File: druid-derive/Cargo.toml [hunk 1]
```diff
@@ -17,3 +17,4 @@ proc-macro2 = "1.0.9"
 
 [dev-dependencies]
 druid = { path = "../druid", version = "0.6.0" }
+float-cmp = "0.6.0"
```
Comment (Finnerale) [inline druid-derive/Cargo.toml:20]: Can we avoid the `num-traits` dependency?
```suggestion
float-cmp = { version = "0.6.0", default-features = false }
```

File: druid/Cargo.toml [hunk 2]
```diff
@@ -47,3 +47,6 @@ version = "0.4.0"
 
 [target.'cfg(target_arch="wasm32")'.dependencies]
 console_log = "0.1.2"
+
+[dev-dependencies]
+float-cmp = "0.6.0"
```
Comment (Finnerale) [inline druid/Cargo.toml:52]: Same as above
```suggestion
float-cmp = { version = "0.6.0", default-features = false }
```

File: druid-derive/Cargo.toml (hunk 1 above)
Comment (xStrom) [reply druid-derive/Cargo.toml]: Sure, we don't need it for this PR.

**Comment by xStrom** [comment]
I reduce the dependency graph!

**Pull request merged**